solr/solrj/src/java/org/apache/solr/client/solrj/io/sql/DatabaseMetaDataImpl.java

```diff
@@ -118,7 +118,7 @@ public String getDatabaseProductVersion() throws SQLException {
     SolrClient solrClient = null;
     for (String node : liveNodes) {
       try {
-        String nodeURL = ZkStateReader.from(cloudSolrClient).getBaseUrlForNodeName(node);
+        String nodeURL = Utils.getBaseUrlForNodeName(node, /*getClusterProperty(URL_SCHEME, "http")*/ "http");
```

Comment:
That's "nocommit" worthy :-)

I suppose CloudSolrClient.getClusterStateProvider should have a getBaseUrlForNodeName() instance method.  The ZK implementation could call the one on ZkStateReader so that this scheme is consulted.